src/gendersprachekorrigieren.ts

```diff
@@ -103,92 +138,103 @@ export class BeGone {
         this.mtype = message.type;
         if (this.currentPageNotExcludedByWhitelistOrBlackList()) {
             //Entfernen bei erstem Laden der Seite
-            this.entferneInitial();
+            this.entferneInitial(document);
 
             //Entfernen bei Seitenänderungen
-            try {
-                const observer = new MutationObserver((mutations: MutationRecord[]) => {
-                    // Der changeAllowedChecker muss geupdated werden bevor entferneInserted(.) aufgerufen wird
-                    this.changeAllowedChecker.handleMutations(mutations);
-
-                    let insertedNodes = new Array<CharacterData>();
-                    mutations.forEach((mutation: MutationRecord) => {
-                        for (let i = 0; i < mutation.addedNodes.length; i++) {
-                            insertedNodes = insertedNodes.concat(this.textNodesUnder(mutation.addedNodes[i]));
-                        }
-                    });
-                    this.entferneInserted(insertedNodes);
-                });
-                observer.observe(document, {
-                    childList: true,
-                    subtree: true,
-                    // attributes needed for changeAllowedChecker
-                    attributes: true,
-                    characterData: false
-                });
-            } catch (e) {
-                console.error(e);
-                chrome.runtime.sendMessage({
-                    action: 'error',
-                    page: document.location.hostname,
-                    source: 'gendersprachekorrigieren.js',
-                    error: e
-                } as ErrorRequest);
-            }
+            this.installMutationObserver(document);
         }
     }
 
 
+    private installMutationObserver(doc: Document) {
```

Comment:
Ausgelagert in der Methode, und SuperPowerfulMutationObserver genutzt. Ansonsten gleich.